### DISCUSSION TITLE
Module config.py's `create_ssl_context` ssl_version param validate to int alone

### DISCUSSION THREAD

**Discussion by Murugaraju** [Kludex/uvicorn#2347] [Q&A]
### Description of problem

Hey Team, 

I had some settings for the gunicorn config in saved in cfg file. First scenario is the execution of gunicorn is done via shell file, the file containing the code as below

```shell
exec su -p randomuser -c 'gunicorn --ssl-version=5 --certfile=../sample/server.crt --keyfile=../sample/server.key  -c app/gunicorn.cfg app.__main__:app'
```
During above scenario the execution of gunicorn done via the worker `UvicornWorker`. The problem found is as below

![image](https://github.com/encode/uvicorn/assets/32738182/e73e898e-3ffb-4f9a-b1d0-886bd67d298c)

Second I moved the `--ssl-version=5` from command arg to cfg file as below

```cfg

ssl_version = 5

```

it worked.

https://github.com/encode/uvicorn/blob/22873a99188413332df98c04a351e061672cb523/uvicorn/config.py#L106

Any chance that this explicit annotation to ssl_verification can be removed. Thought of giving the team a visibility on the scenario I faced 😅

**Comment by frobnitzem** [comment]
In using uvicorn with SSL, I've noticed a few of these kinds of issues.
I think the solution should be to require these inputs have the correct enum type defined by python's ssl module:

* [PROTOCOL values](https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_TLS)
* [VerifyMode](https://docs.python.org/3/library/ssl.html#ssl.VerifyMode)

For example, the uvicorn documentation just says verifymode is an int, but not that verifymode 1 makes client certificates optional.

Using strings would make these choices much easier than integers.
